dms_field/static/src/js/base/dms_tree_renderer.js

```diff
@@ -0,0 +1,460 @@
+                    parent: data.parent,
+                });
+            });
+            this.$('[data-toggle="tooltip"]').tooltip();
+        },
+        _treeChanged: function(ev) {
+            var data = ev.data.data;
+            if (data.selected && data.selected.length === 1) {
+                this._preview_node(data.node);
+            }
+        },
+        start: function() {
+            var res = this._super.apply(this, arguments);
+            this.$tree = this.$(".dms_tree");
+            this.$tree.jstree(this.config);
+            this.start_tree_triggers();
+            return res;
+        },
+        /*
+            This is used to check that the operation is allowed
+        */
+        _checkCallback: function(operation, node, parent) {
+            if (operation === "copy_node" || operation === "move_node") {
+                // Prevent moving a root node
+                if (node.parent === "#") {
+                    return false;
+                }
+                // Prevent moving a child above or below the root
+                if (parent.id === "#") {
+                    return false;
+                }
+                // Prevent moving a child to a settings object
+                if (parent.data && parent.data.model === "dms.storage") {
+                    return false;
+                }
+                // Prevent moving a child to a file
+                if (parent.data && parent.data.model === "dms.file") {
+                    return false;
+                }
+            }
+            return true;
+        },
+        _checkSelect: function(node) {
+            if (this.params.filesOnly && node.data.model !== "dms.file") {
+                return false;
+            }
+            return !(node.parent === "#" && node.data.model === "dms.storage");
+        },
+        _preview_node: function(node) {
+            var $buttons = this.$el.find(".o_dms_extra_actions");
+            $buttons.empty();
+            if (
+                node.data &&
+                ["dms.directory", "dms.file"].indexOf(node.data.model) !== -1
+            ) {
+                this.$(".dms_document_preview").html(
+                    $(
+                        QWeb.render("dms.DocumentTreeViewDirectoryPreview", {
+                            widget: this,
+                            dms_object: node.data,
+                        })
+                    )
+                );
+                var self = this;
+                var menu = this._loadContextMenu(node);
+                _.each(menu, function(action) {
+                    self._generateActionButton(node, action, $buttons);
+                });
+            }
+        },
+        _generateActionButton: function(node, action, $buttons) {
+            if (action.action) {
+                var $button = $("<button>", {
+                    type: "button",
+                    class: "btn btn-secondary " + action.icon,
+                    "data-toggle": "dropdown",
```

Comment:
Here it would be great to add `"title": action.label` to get button information on hovering

Reply:
Fixed :wink: